### PR TITLE
show shell podcast modal after button click

### PR DESCRIPTION
**Pull request by chumomega**
## What does this PR do?
it turns all the items in the list into clickable buttons. i also created a podcast page component to show a podcast

## How can others know that your changes do what you say they do? Can you please share some testing steps?
run app.
observe that there are buttons for the items in the list. 
observe that when you click on button, a modal pops up
observe that you can close modal with close buttons
observe no information is shown in modal

##screenshots

Screen Shot 2022-02-15 at 11 03 31 PM

## Checklist before merging
- [x] Someone has given this PR an approval and thorough review
- [x] I have validated that I have not broken something else in the code

### CODE REVIEW

File: pranker_frontend/src/PodcastList.js [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
     constructor(props) {
         super(props);
 
         this.state = {
-          podcasts: this.props.podcasts
+          show_podcast_modal: false,
+          podcast_to_show: null
         };
+
+        this.button_component = React.createRef();
```
Comment (kwen0) [inline pranker_frontend/src/PodcastList.js:14]: what does this do? I've never worked with refs in react before

File: pranker_frontend/src/components/PodcastModal.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        return (
+            <Modal show={this.state.show} onHide={this.handleClose}>
+                <Modal.Header closeButton>
+                    {/* <Modal.Title>{podcast.name} {podcast.episodeTitle}</Modal.Title> */}
+                </Modal.Header>
+                <Modal.Body>
+                    <Form>
+                        <Form.Group className="mb-3" controlId="formPodcastName">
+                            {/* <Form.Label>{podcast.name} {podcast.episodeTitle}</Form.Label> */}
+                            <Form.Control type="text" placeholder="Enter podcast title" />
```
Comment (kwen0) [inline pranker_frontend/src/components/PodcastModal.js:40]: will this be edited later to show podcast details?

**Review by kwen0 — APPROVED** [review]
looks good! i've forgotten about the bind and this keyword that are used in class components, was good practice to look at it again

File: pranker_frontend/src/PodcastList.js (hunk 1 above)
Comment (chumomega) [reply pranker_frontend/src/PodcastList.js:14]: they basically let you access DOM nodes or react elements created in the render method
i'm no expert on them and found it as a solution to what i was trying to do which was update the state in the child component from the parent component
https://reactjs.org/docs/refs-and-the-dom.html

**Pull request merged**